src/com/assignment2/KaprekerConstant.java

```diff
@@ -0,0 +1,28 @@
+package com.assignment2;
+
+public class KaprekerConstant {
+
+	public static boolean checkKaprekersConstant(int number) {
+		int ascendingSortedNumber = NumberManipulator.ascendingSort(number);
+		int descendingSortedNumber = NumberManipulator.descendingSort(ascendingSortedNumber);
+
+		int difference;
+		int iterationCount = 0;
+		boolean isKaprekersConstant = false;
+
+		while (iterationCount < Constants.MAX_DIFFERENCE_ITERATION) {
+			difference = Math.abs(ascendingSortedNumber - descendingSortedNumber);
+			iterationCount++;
+			System.out.println("\n********\nAscending: " + ascendingSortedNumber + "\niteration: " + iterationCount
+					+ "\nDescending: " + descendingSortedNumber + "\nDifference: " + difference + "\n*********");
+
+			ascendingSortedNumber = NumberManipulator.ascendingSort(difference);
+			descendingSortedNumber = NumberManipulator.descendingSort(ascendingSortedNumber);
+			if (difference == Constants.KAPREKAR_CONSTANT) {

```

Comment:
can we check this if (difference == Constants.KAPREKAR_CONSTANT) just above the line 18 as the no. = karprekar Constant than there will be no need of calling ascending and descending sort function